### PR TITLE
Enhance CLI with reusable prompt functions, avoid code repetition, improve repository creation commands, and prepare changes for version 0.3.0

### PR DESCRIPTION
**Pull request by RichardIrala**
Solution for this issue https://github.com/0xsequence/issue-tracker/issues/3311

### CODE REVIEW

File: src/utils/index.ts [hunk 1]
```diff
@@ -7,4 +8,21 @@ export function isValidPrivateKey(privateKey: string) {
     } catch (error) {
         return false;
     }
+}
+
+export async function promptUserKeyCustomizationDecision () : Promise<boolean> {
+    return await select({
+        message:
+          "Would you like to use test configuration keys for testing, or customize your configuration keys?",
```
Comment (taylanpince) [inline src/utils/index.ts]: Let's change slightly to "Would you like to use default configuration keys for testing..."

File: src/marketplace/create_marketplace_boilerplate.ts [hunk 2]
```diff
@@ -12,101 +13,77 @@ export async function createMarketplaceBoilerplate(program: Command, options: an
...
-        console.log("To skip and use the default value: 'waas', press enter.");
- 
-        walletType = await input({
-            message: "Wallet Type:",
+        walletType = await select({
+            message:
+                "Please provide the Wallet Type for your project.\nYou can use 'waas' for Embedded Wallet and 'universal' for Universal Wallet.\nFor more information on wallet types: https://docs.sequence.xyz/solutions/wallets/overview",
+            choices: [
+                {
+                    name: "waas",
```
Comment (taylanpince) [inline src/marketplace/create_marketplace_boilerplate.ts]: Let's make the name "Embedded Wallet"

File: src/marketplace/create_marketplace_boilerplate.ts [hunk 3]
```diff
@@ -12,101 +13,77 @@ export async function createMarketplaceBoilerplate(program: Command, options: an
...
+                    value: "waas",
+                },
+                {
+                    name: "universal",
```
Comment (taylanpince) [inline src/marketplace/create_marketplace_boilerplate.ts]: Name should be "Universal Wallet"

File: src/boilerplates/create_embedded_wallet_react.ts [hunk 4]
```diff
@@ -14,52 +15,56 @@ export async function createEmbeddedWalletReact(program: Command, options: any)
...
- 
-        appleClientId = await input({
-            message: "Apple Client ID:",
-        });
-
-        console.log("");
+    const userWantsToConfigureTheirKeys = await promptUserKeyCustomizationDecision();
+    
+    if (userWantsToConfigureTheirKeys) {
+        if (!waasConfigKey) {
```
Comment (taylanpince) [inline src/boilerplates/create_embedded_wallet_react.ts]: It feels like we are repeating these `waasConfigKey`, `projectAccessKey` checks and `await input` implementations all over the place. Can you centralize these into a single point as well? That way the implementation will be a lot cleaner.

**Review by taylanpince — CHANGES_REQUESTED** [review]
Minor changes needed, looking great

File: src/marketplace/create_marketplace_boilerplate.ts (hunk 2 above)
Comment (RichardIrala) [reply src/marketplace/create_marketplace_boilerplate.ts]: Done

File: src/marketplace/create_marketplace_boilerplate.ts (hunk 3 above)
Comment (RichardIrala) [reply src/marketplace/create_marketplace_boilerplate.ts]: Done

File: src/utils/index.ts (hunk 1 above)
Comment (RichardIrala) [reply src/utils/index.ts]: Done. Modified to: "Would you like to use the default configuration keys for testing?"

File: src/boilerplates/create_embedded_wallet_react.ts (hunk 4 above)
Comment (RichardIrala) [reply src/boilerplates/create_embedded_wallet_react.ts]: About that can you provide me some feedback about this? https://github.com/0xsequence/sequence-cli/pull/11/files#diff-0e4c4ef1bf4917d07d28ca11950aa794bab7500434ff881ba77fee6630501646. I want to know if this satisfies your idea and if it seems well structured to you.

Comment (taylanpince) [reply src/boilerplates/create_embedded_wallet_react.ts]: @RichardIrala Very good start. Now let's go one step further and do wrapper utility functions that use `promptForKeyWithLogs` under the hood, but specific to most common repeated cases. I suspect those will be `promptForWaaSConfigKeyWithLogs`, `promptForProjectAccessKeyWithLogs` etc.

Comment (RichardIrala) [reply src/boilerplates/create_embedded_wallet_react.ts]: Done

**Comment by RichardIrala** [comment]
# Changes

* Created a new file for types
* Improved some CLI commands by adding missing environment variable prompts
* Added a basic configuration for .prettierrc
* Introduced new functions in the utils folder

## New functions in the utils folder:

* `promptUserKeyCustomizationDecision`
* `promptForKeyWithLogs`
* `promptForProjectAccessKeyWithLogs`
* `promptForWaaSConfigKeyWithLogs`
* `promptForGoogleClientIdWithLogs`
* `promptForAppleClientIdWithLogs`
* `promptForWalletConnectIdWithLogs`
* `writeToEnvFile`
* `writeDefaultKeysToEnvFileIfMissing`

**Review by taylanpince — APPROVED** [review]
This is beautiful – Thank you @RichardIrala

**Pull request merged**